Src/IronPython/Compiler/Parser.cs

```diff
@@ -951,13 +951,23 @@ public class Parser : IDisposable { // TODO: remove IDisposable
             Eat(TokenKind.KeywordExec);
             var start = GetStart();
             Expression code, locals = null, globals = null;
+
             code = ParseExpr();
-            if (MaybeEat(TokenKind.KeywordIn)) {
+            if (code is TupleExpression t) {
```

Comment:
I'm pretty sure this needs to be done the other way around (like in my example) otherwise you get a `SyntaxError` instead of a `TypeError`:
```Python
exec ("",{}) in {}
```